File: java/reverse_string.java
import java.util.Scanner;

public class reverse_string{
    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);

        System.out.print("Enter a string: ");
        String string = sc.next();

        System.out.println("The string is: " + string);
 
        String reversed = "";

        for(int i=0;i<string.length();i++){
            reversed = string.charAt(i) + reversed;
        }
        System.out.println("Reversed string: " + reversed);
    }
}

File: C/reverse_string.c
#include<stdio.h>
int main(){
    char str[100], rev[100];
    int i, j, len = 0;

    printf("Enter a string: ");
    scanf("%s", str);

    while (str[len] != '\0') {
        len++;
    }

    for (i = len - 1, j = 0; i >= 0; i--, j++) {
        rev[j] = str[i];
    }
    rev[j] = '\0'; // Null-terminate the reversed string

    printf("Reversed string: %s\n", rev);

    return 0;   
}